app/models/miq_provision/automate.rb

```diff
@@ -2,17 +2,18 @@ module MiqProvision::Automate
   extend ActiveSupport::Concern
 
   module ClassMethods
-    def get_domain_details(domain, with_password = false, user = nil)
+    def get_domain_details(domain, with_password, user)
       _log.info "<< domain=<#{domain}> with_password=#{with_password} user=<#{user}>"
       if domain.nil?
         _log.error "Domain Not specified"
         return nil
       end
 
-      attrs = {'request' => 'UI_PROVISION_INFO', 'message' => 'get_domains'}
-      attrs[MiqAeEngine.create_automation_attribute_key(user)] = MiqAeEngine.create_automation_attribute_value(user) unless user.nil?
+      attrs = MiqAeEngine.set_automation_attributes_from_objects(
```

Comment:
@kbrock 
Should we check that the user is not nil here. There is already a check in MiqAeEngine.resolve_automation_object that checks if the user is valid.
raise "User object not passed in" unless user_obj.kind_of?(User)